Выполнение lesson 5

ohikoyami opened the pull request.

ohikoyami committed:
Добавление стандартного файла .gitignore

ohikoyami committed:
Добавление файла с базовыми настройками и действиями браузера

ohikoyami committed:
Добавление картинки для анкеты

ohikoyami committed:
Тест прогон на заполнение анкеты с проверкой коректности итоговой формы

rajlful commented on lesson5/test_form_fill_and_result.py:
```diff
@@ -0,0 +1,39 @@
+    browser.element('#currentAddress').type('st. Skobelevskaya')
+    browser.element('#react-select-3-input').type('raja').press_tab()
+    browser.element('#react-select-4-input').type('jaip').press_tab()
+    browser.element('#submit').click()
```
А как выглядит форма ?
Там уже есть все элементы, и ничего не нужно ждать ?

Обычно просто добавляется ожидание какого-то состояния при клике. А тут все нонстопом

ohikoyami replied:
да, там уже все есть на форме, работает и без ожидания + у selene, как говорили, есть неявное ожидание на постоянке

galsv commented on lesson5/test_form_fill_and_result.py:
```diff
@@ -0,0 +1,39 @@
+    browser.element('[for="gender-radio-2"]').click()
+    browser.element('[for="gender-radio-3"]').click()
+    browser.element('#userNumber').type('9876541234')
+    browser.element('#dateOfBirthInput').click()
```
вот тут я бы рекомендовал выбор даты в отдельный метод вытащить и на вход принимать дату в нужном тебе формате.

galsv commented on lesson5/test_form_fill_and_result.py:
```diff
@@ -0,0 +1,39 @@
+    browser.element('#firstName').type('Daria')
+    browser.element('#lastName').type('Bilenko')
+    browser.element('#userEmail').type('e.mail@dssl.ru')
+    browser.element('[for="gender-radio-1"]').click()
```
вот тут тебе как раз не надо тыкать на все подряд, это как раз и есть скрипт с шагами, а не тест. Суть теста это сымитировать шаги реального пользователя, а не проверить кликабельность. По сути тут уже e2e тест реализуется. И я бы рекомендовал переписать селектор на  browser.element('#genterWrapper').element('input[value="Female"]').click()

galsv commented on lesson5/test_form_fill_and_result.py:
```diff
@@ -0,0 +1,39 @@
+    browser.element('.react-datepicker__month-select option[value="3"]').click()
+    browser.element('.react-datepicker__year-select option[value="2002"]').click()
+    browser.element('.react-datepicker__day.react-datepicker__day--009').click()
+    browser.element('#subjectsInput').type('M').press_tab()
```
Тут стоит что-то по осмысленнее вписать

galsv commented on lesson5/test_form_fill_and_result.py:
```diff
@@ -0,0 +1,39 @@
+    browser.element('.react-datepicker__year-select option[value="2002"]').click()
+    browser.element('.react-datepicker__day.react-datepicker__day--009').click()
+    browser.element('#subjectsInput').type('M').press_tab()
+    browser.element('[for="hobbies-checkbox-1"]').click()
```
я бы тут рекомендовал прям на чек бокс кликать через input элемент, а не на текст. У него и селектор поудобнее

galsv commented on lesson5/test_form_fill_and_result.py:
```diff
@@ -0,0 +1,39 @@
+    browser.element('#subjectsInput').type('M').press_tab()
+    browser.element('[for="hobbies-checkbox-1"]').click()
+    browser.element('[for="hobbies-checkbox-2"]').click()
+    browser.element('[for="hobbies-checkbox-3"]').click()
```
тут тоже не стоит на все кликать, можно упустить момент где у тебя эти чек боксы не работают и просто все данные выбираются

galsv commented on lesson5/test_form_fill_and_result.py:
```diff
@@ -0,0 +1,39 @@
+    browser.element('#react-select-3-input').type('raja').press_tab()
+    browser.element('#react-select-4-input').type('jaip').press_tab()
+    browser.element('#submit').click()
+    assert browser.element(".modal-content").element('table').all('tr').all('td').even.should(have.exact_texts(
```
вот тут достаточно сложная конструкция по получению всех четных столбцов которые сравниваются с текстами, советую в таких моментах оставлять комменты.

galsv commented on lesson5/conftest.py:
```diff
@@ -0,0 +1,15 @@
+def manage_browser():
+    browser.config.base_url = 'https://demoqa.com'  # фикстура - дефолт браузер
+    browser.config.timeout = 2.0
+    browser.open('/automation-practice-form') #подтверждение cookie + открытие браузера
```
вот эту штуку я рекомендую в тест вытащить, потому что тут у тебя только общие настройки должны быть для браузера.

ohikoyami committed:
Убрано открытие браузера

ohikoyami committed:
Вынесена дата в отдельный модуль
Выбор пола одним кликом на необходимый
Предмет вписывается полностью
Клик на чекбокс упрощен
Добавлены комментарии к проверки вывода анкеты в таблице

ohikoyami committed:
Поправлены недочеты и изменен модуль с днями

galsv commented on lesson5/test_form_fill_and_result.py:
```diff
@@ -0,0 +1,47 @@
+    browser.element('#dateOfBirthInput').click()
+    browser.element(f'.react-datepicker__month-select option[value="{month - 1}"]').click() #месяцы идут от 0 до 11
+    browser.element(f'.react-datepicker__year-select option[value="{year}"]').click()
+    browser.element(f'.react-datepicker__day.react-datepicker__day--{day:03d}').click()# day:03d это добавление 0 перед числом в зависимости от его длины
```
тут надо бы отделить переносом строки
то бишь сделать один или несколько Enter

galsv commented on lesson5/test_form_fill_and_result.py:
```diff
@@ -0,0 +1,47 @@
+    browser.element('#firstName').type('Daria')
+    browser.element('#lastName').type('Bilenko')
+    browser.element('#userEmail').type('e.mail@dssl.ru')
+    browser.element('#genterWrapper').element('input[value="Female"]').element('..').click() # выбор элемента с явным указанием гендера
```
А вот это зачем element('..')? Он по конкретному не тыкает разве?

galsv replied:
Поглядел сам, реально не тыкает, почему то не кликабельный

galsv commented on lesson5/test_form_fill_and_result.py:
```diff
@@ -0,0 +1,47 @@
+    browser.element('#react-select-4-input').type('jaip').press_tab()
+    browser.element('#submit').click()
+
+    # проверка корректных данных в итоговой форме:
```
Вот тут слишком даже подробно описала, и когда у тебя такой сплошной текст используй """ """ вместо построчной #

ohikoyami committed:
Применила мелкие правки из рекомендаций в комментариях
